loopy/codegen/bounds.py

```diff
@@ -55,57 +55,57 @@ def get_approximate_convex_bounds_checks(domain, check_inames, implemented_domai
 
 # {{{ on which inames may a conditional depend?
 
-def get_usable_inames_for_conditional(kernel, sched_index):
+def get_usable_inames_for_conditional(kernel, sched_indices):
```

Comment:
If I understand this correctly, this builds a bunch of auxiliary data structures that make usable-iname determination fast, and then it's called ahead of time for all spots where this question could arise.

Assuming that that's right, I disagree with how this is structured. IMO, it would be easier to use if the intermediate structures were stuffed into a class that can then rapidly process these queries.

Reply:
Agreed that using a "fast-query" mechanism will be a better fit for all the operations o'er here. I'll give another attempt here.